### PR TITLE
[SPARK-1564] [Docs] Added Javascript to Javadocs to create badges for tags like :: Experimental ::

### PR DESCRIPTION
**Pull request by deroneriksson**
Modified copy_api_dirs.rb and created api-javadocs.js and api-javadocs.css files in order to add badges to javadoc files for :: Experimental ::, :: DeveloperApi ::, and :: AlphaComponent :: tags

### CODE REVIEW

**Comment by andrewor14** [comment]
ok to test

**Comment by rxin** [comment]
This is amazing!

Do you mind pasting a screenshot?

**Comment by deroneriksson** [comment]
Here's a screenshot:

javadoc-badges-example

File: docs/js/api-javadocs.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    var newHTML = oldHTML.replace(tag, "");
+    $(this).html(newHTML);
+  });
+
+  // Add html badge tags
+  tags.each(function(index) {
+    if ($(this).parent().is('td.colLast')) {
+      $(this).parent().prepend(html);
+    } else if ($(this).parent('li.blockList').parent('ul.blockList').parent('div.description').parent().is('div.contentContainer')) {
+      var contentContainer = $(this).parent('li.blockList').parent('ul.blockList').parent('div.description').parent('div.contentContainer')
```
Comment (andrewor14) [inline docs/js/api-javadocs.js]: these two lines are super long! Could you rewrite them in a way that's easier to read? Maybe add line breaks after each `.parent(...)` block?

**Comment by andrewor14** [comment]
@deroneriksson looks great. Thanks for working on this.

File: docs/css/api-javadocs.css [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* Dynamically injected style for the API docs */
```
Comment (sarutak) [inline docs/css/api-javadocs.css]: Could you add a license header?

**Comment by andrewor14** [comment]
Merging into master. Thanks!

**Pull request closed without merging**